pyro/contrib/gp/models/svgp.py

```diff
@@ -73,34 +77,42 @@ def guide(self):
         Xu = self.get_param("Xu")
 
         # define variational parameters
-        mu_0 = torch.tensor(Xu.data.new(self.num_inducing).zero_(), requires_grad=True)
+        if self.y.dim() == 1:
+            mu_0 = torch.tensor(Xu.data.new(Xu.size(0)).zero_(),
+                                requires_grad=True)
+        else:
+            mu_0 = torch.tensor(Xu.data.new(Xu.size(0), self.y.size(1)).zero_(),
+                                requires_grad=True)
         mu = pyro.param("u_loc", mu_0)
-        unconstrained_Lu_0 = torch.tensor(Xu.data.new(self.num_inducing, self.num_inducing).zero_(),
+        unconstrained_Lu_0 = torch.tensor(Xu.new(Xu.size(0), Xu.size(0)).zero_(),
                                           requires_grad=True)
         unconstrained_Lu = pyro.param("unconstrained_u_tril", unconstrained_Lu_0)
         Lu = transform_to(constraints.lower_cholesky)(unconstrained_Lu)
 
         # TODO: use scale_tril=Lu
-        Ku = Lu.t().matmul(Lu) + self.jitter.expand(self.num_inducing).diag()
-        pyro.sample("u", dist.MultivariateNormal(loc=mu, covariance_matrix=Ku))
+        Ku = Lu.t().matmul(Lu) + self.jitter.expand(Lu.size(0)).diag()
+        # correct event_shape for mu
+        mu_t = mu.t() if mu.dim() == 2 else mu
+        pyro.sample("u", dist.MultivariateNormal(loc=mu_t, covariance_matrix=Ku))
```

Comment:
If `mu_t` is batched, you'll need to either enclose in an `iarange` context
```py
if mu.dim() == 1:
    pyro.sample("u", dist.MultivariateNormal(loc=mu, covariance_matrix=Ku))
else:
    with iarange("u", mu.shape[0]):  # <---- should be minibatch size, did I get this right?
        pyro.sample("u", dist.MultivariateNormal(loc=mu.t(), covariance_matrix=Ku))
```
 or reshape with `.reshape(extra_event_dims=1)`
```py
mu_t = mu.t() if mu.dim() == 2 else mu
pyro.sample("u",
            dist.MultivariateNormal(loc=mu_t, covariance_matrix=Ku)
                .reshape(extra_event_dims=mu.dim() - 1))
```
Since everything is reparameterized here, the two forms are equivalent. If you ever foresee discrete parameters (e.g. a mixture of kernels), then the `iarange` form would be preferred.

Reply:
Thank you! I like the `reshape` style more than `iarange` in this case. :D